modules/oe_subscriptions_digest/tests/src/Functional/UserDigestTestBase.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_subscriptions_digest\Functional;
+
+use Drupal\field\Entity\FieldConfig;
+use Drupal\message_digest\Entity\MessageDigestInterval;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+use Drupal\Tests\oe_subscriptions_anonymous\Trait\SubscriptionsPageTrait;
+use Drupal\user\UserInterface;
+
+/**
+ * Tests the user digest.
+ */
+abstract class UserDigestTestBase extends BrowserTestBase {
+
+  use FlagCreateTrait;
+  use SubscriptionsPageTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'oe_subscriptions_digest',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests the user digest preference.
+   *
+   * @param \Drupal\user\UserInterface $user
+   *   The user for the test.
+   * @param callable $fn_get_path
+   *   Callable that returns the path to visit for the test. It also needs to
+   *   take care of logging the user, or requesting a token in order to allow
+   *   visiting the path.
+   */
+  protected function doTestDigestPreferences(UserInterface $user, callable $fn_get_path): void {
+    // Test notifications frequency user settings.
+    $path = $fn_get_path($user);
+    $this->drupalGet($path);
+    $assert_session = $this->assertSession();
+
+    $select = $assert_session->selectExists('Notifications frequency');
+
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+    ], $this->getOptions($select));
+
+    MessageDigestInterval::create([
+      'id' => 'bi_weekly',
+      'label' => 'Bi-weekly',
+      'interval' => '2 weeks',
+    ])->save();
+    $this->drupalGet($path);
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+      'message_digest:bi_weekly' => 'Bi-weekly',
+    ], $this->getOptions($select));
+
+    MessageDigestInterval::create([
+      'id' => 'monthly',
+      'label' => 'Monthly',
+      'interval' => '1 month',
+    ])->save();
+    $this->drupalGet($path);
+    $this->assertEquals([
+      'Send immediately' => 'Send immediately',
+      'message_digest:daily' => 'Daily',
+      'message_digest:weekly' => 'Weekly',
+      'message_digest:bi_weekly' => 'Bi-weekly',
+      'message_digest:monthly' => 'Monthly',
+    ], $this->getOptions($select));
```

Comment:
This stuff goes in setup method.

Reply:
Do you want to move the intervals creation? I wanted to check that after adding new digest options they appear.